src-built-in/components/toolbar/stores/toolbarStore.js

```diff
@@ -275,6 +295,23 @@ class _ToolbarStore {
 		FSBL.Clients.StorageClient.save({ topic: finsembleWindow.name, key: 'blurred', value: false }, cb);
 	}
 
+	/**
+	 * prompts the user to confirm they want to shutdown finsemble
+	 * @return Promise
+	 * @resolve { err, result }
+	 */
+	confirmCloseToolbar = () => {
+		const dialogParams = {
+			title: "Confirm Shutdown",
+			question: "Do you wish to shut down finsemble?",
+			affirmativeResponseLabel: "Shut down",
+			negativeResponseLabel: "Cancel"
+		};
+		return new Promise(resolve => {
+			FSBL.Clients.DialogManager.open("yesNo", dialogParams, (err, result) => resolve({ err, result }));
```

Comment:
Since this is a new promise being added, we should not resolve with an error. If an error happens, reject. Resolve if successful. This is how promises are *intended* to be used. We have tons of existing problems and hard to understand things because of this practice not being followed.

Starting to guard against this practice now gets us in the habit of doing future promises better initially rather than just waiting for it to be decided to do things right and causing more code-churn later.

Reply:
I agree with Garbee on this one.

Reply:
This was already decided. Never throw and error.  Never reject a promise.  If you can convince Terry otherwise, by all means.